File: sort/pro_2.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        int answer=0;
        Arrays.sort(citations);
        int i=citations.length;
        for(int n :  citations){
            answer = Math.max(answer,Math.min(n,i--));
        }
        return answer;
    }
}

File: dp/pro_2.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> a) {
    int answer = 0;
    vector<int> left(a.size());
    vector<int> right(a.size());
    int minL=a[0];
    for(int i=0; i<a.size(); i++){
        minL=min(minL,a[i]);
        left[i]=minL;
    }
    int minR=a[a.size()-1];
    for(int i=a.size()-1; i>=0; i--){
        minR=min(minR,a[i]);
        right[i]=minR;
    }
    for(int i=0; i<a.size(); i++){
        if(a[i]<=left[i] || a[i]<=right[i]) answer++;
    }
    return answer;
}